src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetCommandHandler.cs

```diff
@@ -100,6 +102,27 @@ public CommandState GetCommandState(TabKeyCommandArgs args)
             return CommandState.Available;
         }
 
+        public CommandState GetCommandState(AutomaticLineEnderCommandArgs args, Func<CommandState> nextCommandHandler)
+        {
+            return nextCommandHandler();
+        }
+
+        public void ExecuteCommand(AutomaticLineEnderCommandArgs args, Action nextCommandHandler, CommandExecutionContext executionContext)
+        {
+            AssertIsForeground();
+            if (AreSnippetsEnabled(args)
+                && args.TextView.Properties.TryGetProperty(typeof(AbstractSnippetExpansionClient), out AbstractSnippetExpansionClient snippetExpansionClient)
+                && snippetExpansionClient.IsFullMethodCallSnippet)
+            {
+                // Commit the snippet. Leave the caret in place, but clear the selection. Subsequent handlers in the
+                // chain will handle the remaining Smart Break Line operations.
+                snippetExpansionClient.CommitSnippet(leaveCaret: true);
+                args.TextView.Selection.Clear();
```

Comment:
trying to decide if i find it clear or not how the differnt args are handed by the different handlers.  I think what's bothering me a bit is how this feels a little copy-pasta wiht the code in SnippetCommandHandler.  I feel like maybe this should be extracted out to a helper method, and then that helper is called by this code, and by the code in SnippetCommandHandler if the user hits `;`

Reply:
also, given that we own AbstractSnippetExpansionClient , and that it points at teh textview (i think),i feel like the selection clearing should just be part of hte committing.

Reply:
⏱ adding some documentation to `CommitSnippet`